src/main/java/net/glowstone/util/mojangson/Mojangson.java

```diff
@@ -0,0 +1,542 @@
+package net.glowstone.util.mojangson;
+
+import net.glowstone.util.mojangson.ex.MojangsonParseException;
+import net.glowstone.util.nbt.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static net.glowstone.util.mojangson.MojangsonToken.*;
+
+public class Mojangson {
+
+    /*
+     * You shall not construct.
+     */
+    private Mojangson() {}
+
+    /**
+     * Detects the Tag type of the Mojangson string, and parses it. Convenience method for other parse methods.
+     * This method will fall back to an IntTag if it could not find an appropriate Tag type, and to String if the value
+     * could not be parsed as an Integer either.
+     * @param mojangson The Mojangson string
+     * @return The parsed NBT Tag
+     * @throws MojangsonParseException if the given Mojangson string could not be parsed.
+     */
+    public static Tag parseTag(String mojangson) throws MojangsonParseException {
```

Comment:
I feel like it would be better if these were named toNBT, toNBTInt, etc. I'm not sure on this though.